### PR TITLE
Clean up 'out of date' banner for few more KFP pages

### PR DESCRIPTION
**Pull request by numerology**
including SDK installation, pipeline parameters, static type check in DSL, and GCP specific feature

### CODE REVIEW

File: content/en/docs/pipelines/sdk/install-sdk.md [hunk 1]
```diff
@@ -82,7 +78,7 @@ pip3 install kfp --upgrade
 ```
 **Note:** If you are not using a virtual environment, such as `conda`, when installing the Kubeflow Pipelines SDK, you may receive the following error:
 ```bash
-ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/local/lib/python3.5/dist-packages/kfp-0.2.0.dist-info'
+ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/local/lib/python3.5/dist-packages/kfp-<version>>.dist-info'
```
Comment (Bobgy) [inline content/en/docs/pipelines/sdk/install-sdk.md:81]: nit: looks like there a typo: `>>`

**Review by Bobgy — COMMENTED** [review]
Thank you for updating these @numerology !
/lgtm

It's a little strange why there are updates to images, can you confirm?

There's a minor comment, and do you want to ask @chensun or @Ark-kun to review?

**Comment by numerology** [comment]
> It's a little strange why there are updates to images, can you confirm?

Looks like I need to merge mainstream into my fork. Let me clean this up. Thanks for noting me!

File: content/en/docs/pipelines/sdk/install-sdk.md (hunk 1 above)
Comment (numerology) [reply content/en/docs/pipelines/sdk/install-sdk.md]: Thanks for catching! Corrected

**Comment by joeliedtke** [comment]
Preview:
https://deploy-preview-2174--competent-brattain-de2d6d.netlify.app/docs/pipelines/sdk/gcp/
https://deploy-preview-2174--competent-brattain-de2d6d.netlify.app/docs/pipelines/sdk/install-sdk/
https://deploy-preview-2174--competent-brattain-de2d6d.netlify.app/docs/pipelines/sdk/parameters/
https://deploy-preview-2174--competent-brattain-de2d6d.netlify.app/docs/pipelines/sdk/static-type-checking/

File: content/en/docs/pipelines/sdk/parameters.md [hunk 2]
```diff
@@ -36,6 +32,8 @@ def my_pipeline(
     my_url: str = 'http://example.com'
 ):
   ...
+  # In the pipeline function body one can use `my_num`, `my_name`, `my_url`
```
Comment (joeliedtke) [inline content/en/docs/pipelines/sdk/parameters.md:35]: Suggested revision (speak directly to the user):

In the pipeline function body, you can use the my_num, my_name, and my_url arguments as PipelineParam objects.

**Review by joeliedtke — CHANGES_REQUESTED** [review]
One small item, otherwise this looks great to me.

File: content/en/docs/pipelines/sdk/static-type-checking.md [hunk 3]
```diff
@@ -141,12 +159,16 @@ when the upstream component generates an output with type "*Float*" and the down
...
+  a = task_factory_a(field_l=12)
+  # For each of the arguments, you can also ignore the types by calling
+  # ignore_type function.
+  b = task_factory_b(
+      field_x=a.outputs['field_n'],
+      field_y=a.outputs['field_o'],
+      field_z=a.outputs['field_m'].ignore_type())
+
 compiler.Compiler().compile(pipeline, 'pipeline.tar.gz', type_check=True)
 ```
```
Comment (8bitmp3) [inline content/en/docs/pipelines/sdk/static-type-checking.md:174]: Suggestion:

- Line 183: To provide more information, making it explicit that it's a Jupyter-based demo, and to improve accessibility, perhaps we can slightly rephrase this to:

```diff
- See [type checking sample]
+ Learn how to define a KubeFlow pipeline with Python DSL and compile the
+ pipeline with type checking: a [Jupyter notebook 
+ demo](https://github.com/kubeflow/pipelines/blob/master/samples/core/dsl_static_type_checking/dsl_static_type_checking.ipynb)
```

Or something along these lines.

WDYT?

**Review by 8bitmp3 — CHANGES_REQUESTED** [review]
Hi @numerology Awesome stuff! Just a few suggestions from my side:

- To provide more information, making it explicit that it's a Jupyter-based demo, and to improve accessibility, perhaps we can slightly rephrase this to:

```diff
- See [type checking sample]
+ Learn how to define a KubeFlow pipeline with Python DSL and compile the
+ pipeline with type checking: a [Jupyter notebook 
+ demo](https://github.com/kubeflow/pipelines/blob/master/samples/core/dsl_static_type_checking/dsl_static_type_checking.ipynb)
```
(Or something along those lines)

- For accessibility and improved phrasing ("~~see more~~" -> "for more information, you can refer to" (assuming it's not mandatory in this guide to go there, hence "you can")):

```suggestion
For more information, you can refer to the guide on [building components and
pipelines](/docs/pipelines/sdk/build-component/#create-a-python-class-for-your-component).
```
(Or something along those lines)

- Across all files in this PR: if you have time, add spacing - blank line separators - between 1) paragraphs and code blocks; 2) headings/(sub)titles and paragraphs.

WDYT?

Thank you!

File: content/en/docs/pipelines/sdk/parameters.md (hunk 2 above)
Comment (numerology) [reply content/en/docs/pipelines/sdk/parameters.md]: Thanks. Done

File: content/en/docs/pipelines/sdk/parameters.md [hunk 4]
```diff
@@ -36,6 +32,8 @@ def my_pipeline(
     my_url: str = 'http://example.com'
```
[the remark this answers is not in the document]
Comment (numerology) [reply content/en/docs/pipelines/sdk/parameters.md:32]: Yes I believe you're right. Done

File: content/en/docs/pipelines/sdk/parameters.md [hunk 5]
```diff
@@ -36,6 +32,8 @@ def my_pipeline(
...
 ):
   ...
+  # In the pipeline function body one can use `my_num`, `my_name`, `my_url`
+  # as pipeline parameter.
 ```
 
 See more in the guide to [building a
```
[the remark this answers is not in the document]
Comment (numerology) [reply content/en/docs/pipelines/sdk/parameters.md]: Done.

File: content/en/docs/pipelines/sdk/static-type-checking.md (hunk 3 above)
Comment (numerology) [reply content/en/docs/pipelines/sdk/static-type-checking.md:183]: SG. Done

File: content/en/docs/pipelines/sdk/install-sdk.md [hunk 6]
```diff
@@ -82,7 +78,7 @@ pip3 install kfp --upgrade
 ```
 **Note:** If you are not using a virtual environment, such as `conda`, when installing the Kubeflow Pipelines SDK, you may receive the following error:
 ```bash
```
[the remark this answers is not in the document]
Comment (numerology) [reply content/en/docs/pipelines/sdk/install-sdk.md]: Done.

**Review by numerology — COMMENTED** [review]
Thanks! @joeliedtke and @8bitmp3 I did another pass. PTAL

**Comment by Bobgy** [comment]
/LGTM
/approve

Thank you @numerology !
Great stuff!

I think we can create a new PR if there are further comments
